src/Persistence_matrix/include/gudhi/Persistence_matrix/chain_vine_swap.h

```diff
@@ -0,0 +1,609 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file chain_vine_swap.h
+ * @author Hannah Schreiber
+ * @brief Contains the @ref Chain_barcode_swap and @ref Chain_vine_swap classes, as well as the
+ * @ref Dummy_chain_vine_swap and @ref Dummy_chain_vine_pairing structures.
+ */
+
+#ifndef PM_CHAIN_VINE_SWAP_H
+#define PM_CHAIN_VINE_SWAP_H
+
+#include <utility>  //std::swap & std::move
+#include <cassert>
+#include <functional> //std::function
+
+#include "chain_pairing.h"
+
+namespace Gudhi {
+namespace persistence_matrix {
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Default death comparator. Simply assumes that two positive paired columns are never swapped. Which is true
+ * for the use case in zigzag persistence for example.
+ * 
+ * @param columnIndex1 First column index.
+ * @param columnIndex2 Second column index.
+ * @return false 
+ */
+constexpr bool _no_G_death_comparator([[maybe_unused]] unsigned int columnIndex1,
```

Comment:
I wanted it to appear because it is used as default value. But I can rename it (the "no G" makes reference to the case where no column in G is ever swapped, where G is as defined in the Zigzag paper). With something more explicit like "unused_death_comparator" or "negative_death_comparator"?